apps/server/src/modules/task/domain/StatusDomainService.ts

```diff
@@ -0,0 +1,57 @@
+import { Submission, Task } from '../entity';
+
+export class StatusDomainService {
+	submissions: Submission[];
+
+	constructor(submissions: Submission[]) {
+		this.submissions = submissions;
+	}
+
+	private isGraded(submission: Submission): boolean {
```

Comment:
what is needed to be known that a submission does not know by itself that it is graded or not?